src/core/reducers/errors.js

```diff
@@ -41,12 +44,23 @@ function getMessagesFromError(error) {
     });
 
     if (apiMessages.length) {
-      messages = apiMessages;
+      data = { ...data, messages: apiMessages };
     } else {
       log.warn('API error response did not contain any messages', error);
     }
+
+    if (
+      error.response.data.detail &&
+      // TODO: switch to checking for a code after
+      // https://github.com/mozilla/addons-server/issues/5122
+      error.response.data.detail === 'Signature has expired.'
+    ) {
+      // In this case, the only way to recover from the error is to
+      // refresh the page. It is impossible to fully avoid this state.
```

Comment:
Is there any reason not to refresh the page for the user in this case? Forms are often saved by browsers but we should test that. Maybe it could be a post-merge new issue/patch.